test/solarwinds_apm/oboe_init_options_test.rb

```diff
@@ -458,4 +459,30 @@
     SolarWindsAPM::OboeInitOptions.instance.re_init
     assert_nil(ENV['OTEL_SERVICE_NAME'])
   end
+
+  describe 'test_determine_oboe_log_type' do
+    it 'expect_default_value_0' do
+      ENV.delete('SW_APM_LOG_FILEPATH')
+      ENV.delete('SW_APM_DEBUG_LEVEL')
+      SolarWindsAPM::OboeInitOptions.instance.re_init
+      options = SolarWindsAPM::OboeInitOptions.instance.array_for_oboe
+      _(options[21]).must_equal 0
+    end
+
+    it 'expect_disabled_if_debug_level_is_neg_one' do
+      ENV.delete('SW_APM_LOG_FILEPATH')
+      ENV['SW_APM_DEBUG_LEVEL'] = '-1'
+      SolarWindsAPM::OboeInitOptions.instance.re_init
+      options = SolarWindsAPM::OboeInitOptions.instance.array_for_oboe
+      _(options[21]).must_equal 4
+    end
+
+    it 'expect_to_file_if_SW_APM_LOG_FILEPATH_present' do
+      ENV['SW_APM_LOG_FILEPATH'] = '/custom/path'
+      ENV.delete('SW_APM_DEBUG_LEVEL')
+      SolarWindsAPM::OboeInitOptions.instance.re_init
+      options = SolarWindsAPM::OboeInitOptions.instance.array_for_oboe
+      _(options[21]).must_equal 2
+    end
+  end
```

Comment:
Can we also have tests verify the log_file_path init setting?

Reply:
To create the actual file, we kind need the c library, which is not doable here because we don't compile liboboe anymore for testing

Reply:
Yes, but how about verify that the init option for log file is set to the correct value based on env var? And if log level is disabled, I'd assume we don't ask liboboe to create a log file, i.e. env var for log file is ignored (not set in init option).

Reply:
Actually I think liboboe itself will ignore the log file option if log type is set to disabled? It would be good to see at least ad-hoc testing results in the Jira description.

Reply:
> Yes, but how about verify that the init option for log file is set to the correct value based on env var? 

Yes, we have test for it. I also create new one under logger test section

> And if log level is disabled, I'd assume we don't ask liboboe to create a log file, i.e. env var for log file is ignored (not set in init option). Actually I think liboboe itself will ignore the log file option if log type is set to disabled?

Yes, if the log_type is not file based, the option will be [invalid](https://github.com/solarwinds-cloud/solarwinds-apm-liboboe/blob/master/liboboe/oboe.cpp#L1410-L1414), and liboboe logger only create the file when the file path is valid value ([here](https://github.com/solarwinds-cloud/solarwinds-apm-liboboe/blob/master/liboboe/logger.cpp#L126-L129))

> It would be good to see at least ad-hoc testing results in the Jira description.

You mean like in testbed? I already did manual testing, it seems work (i.e. no file created if disabled)